contentcuration/contentcuration/frontend/channelEdit/vuex/contentNode/actions.js

```diff
@@ -387,38 +377,11 @@ export function updateContentNode(context, { id, mergeMapFields, ...payload } =
   }
 
   if (mergeMapFields) {
-    for (const mapField of mapFields) {
-      if (contentNodeData[mapField]) {
-        if (mapField === 'categories') {
-          // Reduce categories to the minimal set
-          const existingCategories = Object.keys(node.categories || {});
-          const newCategories = Object.keys(contentNodeData.categories);
-          const newMap = {};
-          for (const category of existingCategories) {
-            // If any of the new categories are more specific than the existing category,
-            // omit this.
-            if (!newCategories.some(newCategory => newCategory.startsWith(category))) {
-              newMap[category] = true;
-            }
-          }
-          for (const category of newCategories) {
-            if (
-              !existingCategories.some(
-                existingCategory =>
-                  existingCategory.startsWith(category) && category !== existingCategory
-              )
-            ) {
-              newMap[category] = true;
-            }
-          }
-        } else {
-          contentNodeData[mapField] = {
-            ...node[mapField],
-            ...contentNodeData[mapField],
-          };
-        }
-      }
-    }
+    contentNodeData = {
+      ...contentNodeData,
+      ...getMergedMapFields(node, contentNodeData),
+    };
+    console.log('contentNodeData', contentNodeData);
```

Comment:
👀

Reply:
Whoops. Just pushed the changes! Thank you!